src/loaders/genome_collection/compute_genome_attribs.py

```diff
@@ -310,6 +321,49 @@ def checkm2(genome_ids, work_dir, source_data_dir, debug, program_threads, batch
     return failed_ids
 
 
+def microtrait(genome_ids, work_dir, source_data_dir, debug, program_threads, node_id, source_file_ext):
+    failed_ids, size = list(), len(genome_ids)
+    print(f'Start executing MicroTrait for {len(genome_ids)} genomes')
+
+    batch_dir = _create_batch_dir(work_dir, 'series', size, node_id)
+    tool_genome_id_map, source_genome_file_map = dict(), dict()
+    # retrieve genomic.fna.gz files
+    fna_files = list()
+    for genome_id in genome_ids:
+        genome_file = _find_genome_file(genome_id, source_file_ext, source_data_dir,
+                                        exclude_file_name_substr=['cds_from', 'rna_from', 'ERR'],
+                                        expected_file_count=1)
+
+        if genome_file:
```

Comment:
Should an error be thrown if there's no genome file?

Reply:
there are a lot of GTDB genomes that doesn't have genome files in NCBI. Do we really want to raise? In that case, none of GTDB collection will run.

Reply:
I was thinking the parser script should warm or generate a file for failed genomes. I put a todo item in each tool functions regarding `failed_ids`.

Reply:
Maybe a CLI param then? For PMI etc it can fail, but for GTDB it just warns

Reply:
Or maybe just in the tool config at the start of the script

Reply:
Thinking about this further - the list of IDs to process come from the directory names in the `<collection>/<load_ver>` directory or a genome IDs file which specifies a subset of those directory names. Why are we creating directories that are (I assume) empty, since there's no assembly data? Shouldn't we just not make those directories?

Reply:
Well maybe. But if we don't create directories for every genome. 1. There will be a lack of synchronization between the genomes in our system and the genomes defined in GTDB metadata. 2. Some genomes may have files for tool A, but not for tool B.

Reply:
For 1, there will be a lack of sync anyway because there's nothing to run the tools on anyway, right? So there will be nothing in the system for those genomes (unless we plan to merge data from the metadata file and the tool output, which right now the calc/parser scripts won't handle unless I'm mistaken).

For 2, are you thinking that we'll want to run some tools starting from data other than an assembly file?

Reply:
On phone call we decided to just not have empty folders. @Tianhao-Gu will make a ticket

Reply:
👍 [src/loaders/genome_collection/compute_genome_attribs.py](https://kbase-jira.atlassian.net/browse/RE2022-147)

Reply:
:+1: